examples/gno.land/r/gnoland/boards2/v1/permissions_default.gno

```diff
@@ -0,0 +1,273 @@
+package boards2
+
+import (
+	"errors"
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/nt/commondao"
+
+	"gno.land/r/sys/users"
+)
+
+// DefaultPermissions manages users, roles and permissions.
+type DefaultPermissions struct {
+	superRole Role
+	dao       *commondao.CommonDAO
+	users     *avl.Tree // string(std.Address) -> []Role
+	roles     *avl.Tree // string(role) -> []Permission
+}
+
+// NewDefaultPermissions creates a new permissions type.
+// This type is a default implementation to handle users, roles and permissions.
+func NewDefaultPermissions(dao *commondao.CommonDAO) *DefaultPermissions {
+	if dao == nil {
+		panic("default permissions require a DAO")
+	}
+
+	dp := &DefaultPermissions{
+		dao:   dao,
+		roles: avl.NewTree(),
+		users: avl.NewTree(),
+	}
+
+	// Make sure that DAO users are indexed when DAO has members
```

Comment:
It appears weak because if the DAO changes, this statement may no longer hold true; it essentially makes sense only at T0. In my opinion, you should eliminate this step by allowing the helpers to accept an uninitialized role.

Reply:
Addressed in [6f1205c](https://github.com/gnolang/gno/pull/2901/commits/6f1205c714b9072c2054f1c1167a573557e8a244)

This step indeed had to be removed, it made no sense to pre fill the users tree.